### ISSUE TITLE
Implement an interface for time

### ISSUE THREAD

**Issue by JohnRoesler** [go-co-op/gocron#10]
```
type Time interface {
    Now() time.Time
    After(d time.Duration) <-chan time.Time
    ...
}

type realTime struct{}
func (realTime) Now() time.Time { return time.Now() }
func (realTime) After(d time.Duration) <-chan time.Time { return time.After(d) }

type fakeTime struct{}
func (fakeTime) Now() time.Time { return ... }
func (fakeTime) After(d time.Duration) <-chan time.Time { return ... }
```

**Comment by arjunmahishi** [comment]
Do we also want `NewTicker` to be a part of this interface?

**Comment by JohnRoesler** [comment]
For sure, i was just providing an example of how I think it should look. Probably more funcs we'd need to add 👍

**Comment by Streppel** [comment]
merged #12!
